crates/wac-parser/src/ast/type.rs

```diff
@@ -1412,6 +1415,165 @@ impl AstDisplay for InlineInterface<'_> {
+#[pest_ast(rule(Rule::WorldRef))]
+pub enum WorldRef<'a> {
+    /// The reference is by identifier.
+    Ident(Ident<'a>),
+    /// The reference is by package path.
+    Path(PackagePath<'a>),
+}
+
+impl AstDisplay for WorldRef<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>, indenter: &mut Indenter) -> fmt::Result {
+        match self {
+            Self::Ident(id) => id.fmt(f, indenter),
+            Self::Path(path) => path.fmt(f, indenter),
+        }
+    }
+}
+
+display!(WorldRef);
+
+/// Represents the `with` clause of the `include` statement.
+#[derive(Debug, Clone, FromPest)]
+#[pest_ast(rule(Rule::WorldIncludeWithClause))]
+pub struct WorldIncludeWithClause<'a> {
+    /// The `with` keyword in the clause.
+    pub keyword: With<'a>,
+    /// The names to import with.
+    pub body: WorldIncludeWithBody<'a>,
+}
+
+impl AstDisplay for WorldIncludeWithClause<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>, indenter: &mut Indenter) -> fmt::Result {
+        write!(f, "{keyword} ", keyword = self.keyword,)?;
+        self.body.fmt(f, indenter)
+    }
+}
+
+display!(WorldIncludeWithClause);
+
+/// Represents the body of the include-with clause.
+#[derive(Debug, Clone, FromPest)]
+#[pest_ast(rule(Rule::WorldIncludeWithBody))]
+pub struct WorldIncludeWithBody<'a> {
+    /// The opening brace of the world body.
+    pub open: OpenBrace<'a>,
+    /// The items of the world body.
+    pub items: IncludeNamesList<'a>,
+    /// The closing brace of the world body.
+    pub close: CloseBrace<'a>,
+}
+
+impl AstDisplay for WorldIncludeWithBody<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>, indenter: &mut Indenter) -> fmt::Result {
+        writeln!(f, "{open}", open = self.open)?;
+        self.items.fmt(f, indenter)?;
+        write!(f, "{indenter}{close}", close = self.close)
+    }
+}
+
+display!(WorldIncludeWithBody);
+
+/// The list of names as part of an include-statement `with` clause.
+#[derive(Debug, Clone, FromPest)]
+#[pest_ast(rule(Rule::IncludeNamesList))]
+pub struct IncludeNamesList<'a> {
+    /// The list of names to be included.
+    pub names: Vec<IncludeNamesItem<'a>>,
+}
+
+impl AstDisplay for IncludeNamesList<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>, indenter: &mut Indenter) -> fmt::Result {
+        indenter.indent();
+        let mut idx = 0usize;
+        for name in &self.names {
+            write!(f, "{indenter}")?;
+            name.fmt(f, indenter)?;
+            if idx == self.names.len() - 1 {
+                writeln!(f)?;
+            } else {
+                writeln!(f, ",")?;
+            }
+            idx += 1;
+        }
+        indenter.dedent();
+        Ok(())
+    }
+}
+
+display!(IncludeNamesList);
+
+/// Represents a renaming of an include name.
+#[derive(Debug, Clone, FromPest)]
+#[pest_ast(rule(Rule::IncludeNamesItem))]
+pub struct IncludeNamesItem<'a> {
+    /// The source name include from a world.
+    pub name: Ident<'a>,
+    /// The `as` keyword.
+    pub as_: As<'a>,
```

Comment:
Nit: perhaps `keyword` like we have in so many other AST structs? One of the reasons I used to generic sounding "keyword" was to avoid having to rename it for Rust-keyword conflicts.